src/pools/D3MCompoundDaiPool.sol

```diff
@@ -0,0 +1,201 @@
+
+    ComptrollerLike public immutable comptroller;
+    TokenLike       public immutable dai;
+    CErc20Like      public immutable cDai;
+
+    // --- Auth ---
+    function rely(address usr) external auth {
+        wards[usr] = 1;
+        emit Rely(usr);
+    }
+    function deny(address usr) external auth {
+        wards[usr] = 0;
+        emit Deny(usr);
+    }
+    modifier auth {
+        require(wards[msg.sender] == 1, "D3MCompoundDaiPool/not-authorized");
+        _;
+    }
+
+    // --- Events ---
+    event Rely(address indexed usr);
+    event Deny(address indexed usr);
+    event File(bytes32 indexed what, address data);
+    event Collect(address indexed king, address indexed comp, uint256 amt);
+
+    constructor(address hub_, address dai_, address cDai_) public {
+        address comptroller_ = CErc20Like(cDai_).comptroller();
+
+        require(comptroller_ != address(0), "D3MCompoundDaiPool/invalid-comptroller");
+        require(dai_         == CErc20Like(cDai_).underlying(), "D3MCompoundDaiPool/cdai-dai-mismatch");
+
+        comptroller = ComptrollerLike(comptroller_);
+        dai         = TokenLike(dai_);
+        cDai        = CErc20Like(cDai_);
+
+        TokenLike(dai_).approve(cDai_, type(uint256).max);
+
+        CanLike(D3mHubLike(hub_).vat()).hope(hub_);
+
+        wards[msg.sender] = 1;
+        emit Rely(msg.sender);
+    }
+
+    // --- Math ---
+    uint256 internal constant WAD = 10 ** 18;
+
+    function _add(uint256 x, uint256 y) internal pure returns (uint256 z) {
+        require((z = x + y) >= x, "D3MCompoundDaiPool/overflow");
+    }
+    function _mul(uint256 x, uint256 y) internal pure returns (uint256 z) {
+        require(y == 0 || (z = x * y) / y == x, "D3MCompoundDaiPool/overflow");
+    }
+    function _wmul(uint256 x, uint256 y) internal pure returns (uint256 z) {
+        z = _mul(x, y) / WAD;
+    }
+    function _wdiv(uint256 x, uint256 y) internal pure returns (uint256 z) {
+        z = _mul(x, WAD) / y;
+    }
+    function _min(uint256 x, uint256 y) internal pure returns (uint256 z) {
+        z = x <= y ? x : y;
+    }
+
+    // --- Administration ---
+    function file(bytes32 what, address data) external auth {
+        if (what == "king") king = data;
+        else revert("D3MCompoundDaiPool/file-unrecognized-param");
+        emit File(what, data);
+    }
+
+    function hope(address hub) external override auth {
+        CanLike(D3mHubLike(hub).vat()).hope(hub);
+    }
+
+    function nope(address hub) external override auth {
+        CanLike(D3mHubLike(hub).vat()).nope(hub);
+    }
+
+    function deposit(uint256 amt) external override auth {
+        uint256 prev = cDai.balanceOf(address(this));
+        require(cDai.mint(amt) == 0, "D3MCompoundDaiPool/mint-failure");
+        require(
+            cDai.balanceOf(address(this)) ==
+            _add(prev, _wdiv(amt, cDai.exchangeRateStored())), "D3MCompoundDaiPool/incorrect-cdai-credit"
```

Comment:
Consider adding a comment with a rational for using `exchangeRateStored`

Reply:
Done.